pallas-txbuilder/src/builder.rs

```diff
@@ -0,0 +1,201 @@
+use indexmap::IndexMap;
+use pallas_codec::utils::Bytes;
+
+use pallas_primitives::babbage::{
+    AddrKeyhash, Certificate, NetworkId, PlutusData, TransactionBody, TransactionInput,
+    TransactionOutput, WitnessSet,
+};
+use pallas_traverse::ComputeHash;
+
+use crate::{
+    asset::MultiAsset,
+    fee::Fee,
+    transaction::{self, OutputExt},
+    NetworkParams, ValidationError,
+};
+
+pub struct TransactionBuilder {
+    inputs: IndexMap<TransactionInput, TransactionOutput>,
+    outputs: Vec<TransactionOutput>,
+
+    reference_inputs: Vec<TransactionInput>,
+    collaterals: Vec<TransactionInput>,
+    collateral_return: Option<TransactionOutput>,
+    network_params: NetworkParams,
+    mint: Option<MultiAsset<i64>>,
+    required_signers: Vec<AddrKeyhash>,
+    valid_after: Option<u64>,
+    valid_until: Option<u64>,
+    certificates: Vec<Certificate>,
+    plutus_data: Vec<PlutusData>,
+}
+
+impl TransactionBuilder {
```

Comment:
Are you considering doing any sort of automatic input selection? Or are you saying that's out-of-scope. I can get behind either answer, I just know CML has that.

Reply:
Out of scope for now, I plan on adding it on a more advanced implementation later, if there's demand for it.